app/admin_components/adf-services/dreamfactory-services.js

```diff
@@ -348,6 +348,45 @@ angular.module('dfServices', ['ngRoute', 'dfUtility'])
                         scope.applyFilter();
                     }
                 });
+
+                var goToSelectedApiDoc = function () {
+                    
+                    if (!scope.dataLoading) {
+                        // We want to seperate out the service name, which can be done by pulling out anything
+                        // after the second hash.
+                        var url = top.window.location.hash,
+                        delimiter = '#',
+                        start = 2,
+                        serviceNameAsArray = url.split(delimiter).slice(start),
+                        selectedServiceName = serviceNameAsArray.toString();
+                        
+                        function isCurrentService(service) {
+                            return service.record.name === selectedServiceName;
+                        }
+                        // We then check the selectedServiceName matches to one of the services in the 
+                        // ApiDocs table. If it does, then we open that Api Doc.
+                        var selectedService = scope.filteredServices.find(isCurrentService);
+                        
+                        if (selectedService) {
+                            scope.editService(selectedService.record);
+                        }
+                    } else {
+                        // Wait until all services have loaded in before going to that API Doc.
+                        setTimeout(goToSelectedApiDoc, 300);
+                    }
+                };
+                
+                var init = function () {
+                    // On pageload, check whether the browser url contains a #{serviceName}, which will occur
+                    // if coming from a link in the services dashboard. If it does, then we want to 
+                    // automatically load the selected API Doc    
+                    var regex = /#.+#.+/;
```

Comment:
what will happen if I put `#apidoc#` so without the service name to the URL?
I think there could be an error in the console. Please cover this case.

Reply:
I am surprised you know how to use regex. Good job!

Reply:
Moved up in to the apidata watcher, that once everythings loaded in will check the url.
As for #apidocs# -> this wont lead to an error as the regex check #.+ is looking for _at least_ one character after the #. No errors in the console.
Thanks!